src/HDF5.jl

```diff
@@ -1973,9 +1996,8 @@ function get_mem_compatible_jl_type(objtype::HDF5Datatype)
             return NamedTuple{Symbol.(membernames), Tuple{membertypes...}}
         end
     elseif class_id == H5T_ARRAY
-        nd = h5t_get_array_ndims(objtype.id)
-        dims = Vector{hsize_t}(undef,nd)
-        h5t_get_array_dims(objtype.id, dims)
+        dims = h5t_get_array_dims(objtype.id)
```

Comment:
I guess we could also just have `h5t_get_array_dims` return a tuple of `(dims,ndims)` but that may not be worth it?

Reply:
That's an interesting idea — I'd been thinking of these helper functions doing just the minimum amount of work required to turn out-arguments into return values. Since there are several cases like this one, though, where the implementation is a sequence of two calls that both have meaning, I suppose you could argue that it should return all relevant return/out values. In that case, I guess I'd naturally sequence it as `(ndims, dims)` since the `ndims` call is the "first return", but I'm not set on that interpretation necessarily.

Reply:
Yeah I prefer the reverse ordering, since it's usually the second call that does the 'important' call. I'm not dead set on changing it, just thought I'd mention that this effectively does more work than necessary since the internal already calculated `ndims`

Reply:
ordering `(dims, ndims)` would also kinda align it with `h5s_get_simple_extent_dims` which returns `(dims, maxdims)`

Reply:
What about the other similar cases (that don't lead to making a `String`) — `h5f_get_obj_ids` and `h5p_get_chunk`? Also, would `h5s_get_simple_extent_dims` in this scheme actually gain a third return value so it'd be `(dims, maxdims, ndims)` for consistency?

Reply:
If we decide to go with this (reverse) ordering, I would argue (in addition to consistency) that there isn't any harm in including them as outputs from a performance an usage standpoint. In most cases, we probably don't need them, anyways, but they are available if needed.

This is also why I don't feel very strongly about changing them to include all outputs, but it does seem 'nice' to use this data as it _is_ after all available.

Reply:
I'm fine with that reasoning. Make the changes in this PR, or another one?

Reply:
Either is fine with me. Is this PR still WIP/RFC?

Reply:
No, not still in progress. I removed the WIP flag but forgot to change the title. I can make the couple changes for this, though, and push for a final run through CI.

Reply:
Ah, test suites! The naive thing to do actually does break a test as written. So maybe it will be better to follow-up in another PR.